lib/ansible/modules/network/ironware/ironware_vlan.py

```diff
@@ -0,0 +1,287 @@
+    for key, value in parsed.iteritems():
+        value = value.strip().splitlines()
+        obj = dict()
+        obj['vlan_id'] = key
+        match = re.match(r'Name ([^\s,]+)', value[0])
+        if match and match.group(1) != '[None]':
+            obj['name'] == match.group(1)
+        else:
+            obj['name'] = None
+        obj['state'] = 'active'
+
+        obj['interfaces'] = list()
+        for l in value:
+            match = re.match(r'^(\d+\/\d+)\s+[A-Z]+\s+(TAGGED|UNTAGGED)\s+', l)
+            if match:
+                obj['interfaces'].append(match.group(1))
+
+        objs.append(obj)
+
+    return objs
+
+
+def map_params_to_obj(module):
+    obj = list()
+    aggregate = module.params.get('aggregate')
+    if aggregate:
+        for item in aggregate:
+            for key in item:
+                if item.get(key) is None:
+                    item[key] = module.params[key]
+
+            d = item.copy()
+            d['vlan_id'] = str(d['vlan_id'])
+
+            obj.append(d)
+    else:
+        obj.append({
+            'vlan_id': str(module.params['vlan_id']),
+            'name': module.params['name'],
+            'state': module.params['state'],
+            'interfaces': module.params['interfaces']
+        })
+
+    return obj
+
+
+def check_declarative_intent_params(want, module):
+    if module.params['interfaces']:
+        time.sleep(module.params['delay'])
+        have = map_config_to_obj(module)
+
+        for w in want:
+            for i in w['interfaces']:
+                obj_in_have = search_obj_in_list(w['vlan_id'], have)
+
+                if obj_in_have and 'interfaces' in obj_in_have and i not in obj_in_have['interfaces']:
+                    module.fail_json(msg="Interface %s not configured on vlan %s" % (i, w['vlan_id']))
+
+
+def main():
+    """ main entry point for module execution
+    """
+    element_spec = dict(
+        vlan_id=dict(type='int'),
+        name=dict(),
+        interfaces=dict(type='list'),
+        delay=dict(default=10, type='int'),
+        state=dict(default='present',
+                   choices=['present', 'absent'])
+    )
+
+    aggregate_spec = deepcopy(element_spec)
+    aggregate_spec['vlan_id'] = dict(required=True)
+
+    # remove default in aggregate spec, to handle common arguments
+    remove_default_spec(aggregate_spec)
+
+    argument_spec = dict(
+        aggregate=dict(type='list', elements='dict', options=aggregate_spec),
+        purge=dict(default=False, type='bool')
+    )
+
+    argument_spec.update(element_spec)
+    argument_spec.update(ironware_argument_spec)
+
+    required_one_of = [['vlan_id', 'aggregate']]
+    mutually_exclusive = [['vlan_id', 'aggregate']]
+    module = AnsibleModule(argument_spec=argument_spec,
+                           supports_check_mode=True,
+                           required_one_of=required_one_of,
+                           mutually_exclusive=mutually_exclusive)
+
+    check_args(module)
```

Comment:
`check_args()` is a empty function. Is this call required? For other networks platforms `check_args()` is present for legacy reason.

Reply:
removed